app/src/main/java/io/gnosis/safe/ui/transactions/TransactionListViewModel.kt

```diff
@@ -38,9 +32,19 @@ class TransactionListViewModel
     appDispatchers: AppDispatchers
 ) : BaseStateViewModel<TransactionsViewState>(appDispatchers) {
 
-    init {
+    private lateinit var listType: TransactionListFragment.Type
+
+    fun loadHistoryAndSubscribeToSafeChanges() {
```

Comment:
you could call viewModel.load in fragment's state observer (is ActiveStateChanged) and use type flag of the fragment. In init block of the viewModel you just inform about safe change:
```kotlin
init {
        safeLaunch {
            safeRepository.activeSafeFlow().collect { 
 updateState { TransactionsViewState(isLoading = true, viewAction = iActiveSafeChanged(safe)  }
 }
        }
    }
```

Reply:
I'd prefer to have `load(type, safeChange)` over using 
```
if (type == TransactionPagingSource.Type.HISTORY) {
                viewModel.loadHistory()
            } else {
                viewModel.loadQueue()
            }
```

Reply:
See: fd361c4b17d3c7ce44d9407b2fcc07cdffdd3401

Reply:
looks good